### PR TITLE
Initial rector integration

### PR DESCRIPTION
**Pull request by jordisala1991**

### CODE REVIEW

File: templates/project/.php-cs-fixer.dist.php [hunk 1]
```diff
@@ -6,6 +6,17 @@
  * It's auto-generated by sonata-project/dev-kit package.
  */
 
+declare(strict_types=1);
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:9]: In tests/bootstrap.php, we have the following order
image

- strict
- header
- uses
- DO NOT EDIT THIS FILE

In lot of other files, we have header before the `do not edit this file` message.

I would say we should always use the same order
- declare strict
- header
- Do not edit this file
- uses and php code

**Review by VincentLanglet — CHANGES_REQUESTED** [review]

**Comment by jordisala1991** [comment]
One thing I wanted to consider is how much dependencies rector has, and it is only dependant on php and phpstan, so it is a pretty safe addition to our composer.json (https://github.com/rectorphp/rector/blob/main/composer.json)

File: templates/project/.php-cs-fixer.dist.php (hunk 1 above)
Comment (jordisala1991) [reply templates/project/.php-cs-fixer.dist.php:9]: Agree and fixed on every php file.

File: src/Domain/Value/Project.php [hunk 2]
```diff
@@ -74,8 +75,8 @@ private function __construct(
         array $phpunitExtensions,
         bool $hasDocumentation,
         bool $hasTestKernel,
+        bool $rector,
         ?string $customDoctorRstWhitelistPart,
-        ?string $documentationBadgeSlug
```
Comment (OskarStark) [inline src/Domain/Value/Project.php:78]: Why remove this?

Comment (jordisala1991) [reply src/Domain/Value/Project.php:78]: Fixed.

File: src/Domain/Value/Project.php [hunk 3]
```diff
@@ -123,6 +125,7 @@ public static function fromValues(string $name, array $config, Package $package)
             $phpunitExtensions,
             $config['has_documentation'],
             $config['has_test_kernel'],
+            $config['rector'],
             $config['custom_doctor_rst_whitelist_part'],
             $config['documentation_badge_slug'],
```
[the remark this answers is not in the document]
Comment (jordisala1991) [reply src/Domain/Value/Project.php:131]: Fixed.

**Review by VincentLanglet — APPROVED** [review]

**Comment by jordisala1991** [comment]
I will wait before the merge to solve the admin bundle PR.

**Pull request merged**